### PR TITLE
React-ui: Exposed status prop in useAuth

### PR DESCRIPTION
**Pull request by jmderby**
## Description

Expose `status` prop in `useAuth` hook to enable better loading UX in applications. This change allows developers to create more responsive interfaces by providing granular control over authentication states. Also added test case for new prop.
## Test plan

Ensure demo still works as usual and all loading UX is suffice.

## Package updates

**@crossmint/client-sdk-react-ui** 
- added prop 'status' to useAuth hook
- updated test case

### CODE REVIEW

File: packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx [hunk 1]
```diff
@@ -101,12 +96,13 @@ describe("CrossmintAuthProvider", () => {
             embeddedWallets,
         });
 
-        expect(getByTestId("status").textContent).toBe("in-progress");
+        expect(getByTestId("wallet-status").textContent).toBe("in-progress");
```
Comment (devlyn37) [inline packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx:99]: should probably check the initial auth status here too

File: packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx [hunk 2]
```diff
@@ -46,23 +46,18 @@ function renderAuthProvider({
...
-            <div data-testid="status">{status}</div>
+            <div data-testid="wallet-status">{walletStatus}</div>
+            <div data-testid="auth-status">{authStatus}</div>
             <div data-testid="wallet">{wallet ? "Wallet Loaded" : "No Wallet"}</div>
-            <input
-                data-testid="jwt-input"
-                type="text"
-                onChange={(e) => {
-                    console.log("onChange!");
-                    console.log(e.target.value);
```
Comment (devlyn37) [inline packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx:61]: XD, ty for fixing

**Review by devlyn37 — APPROVED** [review]
nice

File: packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.tsx [hunk 3]
```diff
@@ -52,16 +61,33 @@ export function CrossmintAuthProvider({ embeddedWallets, children, appearance }:
...
         if (crossmint.jwt) {
             document.cookie = `${SESSION_PREFIX}=${crossmint.jwt}; path=/;SameSite=Lax;`;
         }
     }, [crossmint.jwt]);
 
+    const getAuthStatus = (): AuthStatus => {
+        if (crossmint.jwt != null) {
+            return "logged-in";
+        }
+        if (modalOpen === true) {
```
[the remark this answers is not in the document]
Comment (jmderby) [reply packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.tsx:76]: good catch

File: packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx (hunk 2 above)
Comment (jmderby) [reply packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx:61]: np XD

File: packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx (hunk 1 above)
Comment (jmderby) [reply packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx:99]: done

**Pull request merged**